docs/design/2290-cadence-ndc.md

```diff
@@ -0,0 +1,310 @@
+| -------- | ------------- | --------------- | ------- |
+| 1        | 1             | 2               | 1       |
+| 2        | 1             |                 |         |
+| -------- | ------------- | --------------- | ------- |
+```
+
+T = 2:  adding event with event ID == 3 & version == 1
+```
+| -------- | ------------- | --------------- | ------- |
+| Events                   | Version History           |
+| -------- | ------------- | --------------- | ------- |
+| Event ID | Event Version | Event ID        | Version |
+| -------- | ------------- | --------------- | ------- |
+| 1        | 1             | 3               | 1       |
+| 2        | 1             |                 |         |
+| 3        | 1             |                 |         |
+| -------- | ------------- | --------------- | ------- |
+```
+
+T = 3:  domain failover triggered, domain version is now 2
+        adding event with event ID == 4 & version == 2
+```
+| -------- | ------------- | --------------- | ------- |
+| Events                   | Version History           |
+| -------- | ------------- | --------------- | ------- |
+| Event ID | Event Version | Event ID        | Version |
+| -------- | ------------- | --------------- | ------- |
+| 1        | 1             | 3               | 1       |
+| 2        | 1             | 4               | 2       |
+| 3        | 1             |                 |         |
+| 4        | 2             |                 |         |
+| -------- | ------------- | --------------- | ------- |
+```
+
+T = 4:  adding event with event ID == 5 & version == 2
+```
+| -------- | ------------- | --------------- | ------- |
+| Events                   | Version History           |
+| -------- | ------------- | --------------- | ------- |
+| Event ID | Event Version | Event ID        | Version |
+| -------- | ------------- | --------------- | ------- |
+| 1        | 1             | 3               | 1       |
+| 2        | 1             | 5               | 2       |
+| 3        | 1             |                 |         |
+| 4        | 2             |                 |         |
+| -------- | ------------- | --------------- | ------- |
+```
+
+Since Cadence is AP, during failover (change of active data center of a domain), there exist case that more than one data center can modify a workflow, causing divergence of workflow history. Below shows how version history will look like under such condition.
+
+Example, version history with data conflict:
+
+Below will show version history of the same workflow in 2 different data centers.
+
+T = 0:  existing version history in data center A & B
+```
+| -------- | ------------- | --------------- | ------- |
+| Events                   | Version History           |
+| -------- | ------------- | --------------- | ------- |
+| Event ID | Event Version | Event ID        | Version |
+| -------- | ------------- | --------------- | ------- |
+| 1        | 1             | 2               | 1       |
```

Comment:
Might be easier to understand by use the real version. Something like data center A version: 1, data center B version: 2. version increment = 3.